### PR TITLE
Implement minimal DAL

### PR DESCRIPTION
**Pull request by fedyafedko**

### CODE REVIEW

File: StudyHub/StudyHub.DAL/Repositories/Repository.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
+﻿using Microsoft.EntityFrameworkCore;
+using StudyHub.DAL.EF;
+using StudyHub.DAL.Repositories.Interfaces;
+using System.Collections;
+using System.Linq.Expressions;
+
+namespace StudyHub.DAL.Repositories;
+public class Repository<TEntity> : IRepository<TEntity>
+    where TEntity : class
```
Comment (aquaprogit) [inline StudyHub/StudyHub.DAL/Repositories/Repository.cs:9]: maybe better to have `TEntity : EntityBase` to prevent using any class with repository?

**Review by aquaprogit — CHANGES_REQUESTED** [review]

**Comment by aquaprogit** [comment]
Implement EntityBase and inherit all entities from this abstract class to use it as generic in Repos

File: StudyHub/Entities/OpenEndedOption.cs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+﻿namespace StudyHub.Entities;
+
+public class OpenEndedOption : AssignmentTaskOptionBase
+{
```
Comment (aquaprogit) [inline StudyHub/Entities/OpenEndedOption.cs:4]: Use common code style for all classes to keep code style consistent
_Use style like Admin class uses_

File: StudyHub/StudyHub.DAL/Repositories/Repository.cs [hunk 3]
```diff
@@ -0,0 +1,128 @@
+﻿using Microsoft.EntityFrameworkCore;
+using StudyHub.DAL.EF;
+using StudyHub.DAL.Repositories.Interfaces;
+using StudyHub.Entities;
+using System.Collections;
+using System.Linq.Expressions;
+
+namespace StudyHub.DAL.Repositories;
+public class Repository<TEntity> : IRepository<TEntity>
+    where TEntity : EntityBase
```
Comment (aquaprogit) [inline StudyHub/StudyHub.DAL/Repositories/Repository.cs:10]:
```suggestion
    where TEntity : class
```

**Review by aquaprogit — CHANGES_REQUESTED** [review]

**Review by aquaprogit — APPROVED** [review]

**Review by fedyafedko — COMMENTED** [review]
update style

**Pull request merged**